### DISCUSSION TITLE
.NET integration into Windows

### DISCUSSION THREAD

**Discussion by AnakinRaW** [dotnet/runtime#46363] [Ideas] [answered]
Hey .NET team. 
I was wondering whether there were any plans on your side to integrate a .NET runtime into a Windows installation?

The main advantage I see here is that .NET applications work out of the box with a clean Windows installation without a user having to install anything. That's something I really liked for .NET Framework (compared e.g. to Java). 
The "windows native runtime" then could also make use from Windows Updates.

I know that .NET supports bundled apps that come along with their own runtime but for some users it might feel ill to download a 40-70MB for an application which, if developed for .net framework, would only be 1-3MB large. 

I'd like to hear your thoughts on that. 
Cheers

**Comment by GeraudFabien** [comment, marked as answer]
Hello,
Somethinks like that https://devblogs.microsoft.com/dotnet/net-core-updates-coming-to-microsoft-update/ ?

**Reply by AnakinRaW** [reply]
Hey thanks. 

That's a great article describing how everything will behave once .net is installed on the machine. However I'm still missing their information/thought if we get some version deployed along with Windows. 

I get the point where Jamshed .NET is not part of the Windows OS. So it seems like they want to have a clear border between OS critical things and other products (like .net). I guess, adding some .net runtime to windows would create some kind of "contract" they have to maintain and opens questions like which version of .net will be shipped with windows?, (when) do we update its major release?, etc.
I'm just interested whether we still could expect that to happen or of they intentionally decided against from doing this.

**Reply by danmoseley** [reply]
I’m not aware of any plans to ship .NET Core as part of the Windows OS. That would reintroduce many challenges of .NET Framework eg disk space vs in place compatibility, synchronizing schedule and servicing process, etc. 
This release we will continue efforts to make .NET more trimmable for size, which may makes it easier to deploy with apps for those that want to.
